### ISSUE TITLE
Eager macro-expansion error from `lsp-terraform` package when starting `lsp-mode`

### ISSUE THREAD

**Issue by Padraic-O-Mhuiris** [doomemacs/doomemacs#7713]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.
- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

Expected several language modes dependent on lsp to spin up successfully in a given buffer.

### Current behavior

This may not directly be related to how go-mode works but how `lsp-mode` operates. This was reproduced after a fresh install. For whatever reason `lsp-terraform` as defined in `lsp-client-packages` issues a strange error once required correlating to this [line](https://github.com/emacs-lsp/lsp-mode/blob/28b21c800707111bbface28c239ecd105b7e61ff/clients/lsp-terraform.el#L305) which has been unchanged in two years which makes me think that there is a misalignment in doom.

I mutated the files in the built files in `.config/emacs/.local/straight/build-*/lsp-mode/lsp-terraform.el` and new key misalignments occurred between `-` and `_`. 
```
Debugger entered--Lisp error: (error "Eager macro-expansion failure: (error \"Unknown key...")
  error("Eager macro-expansion failure: %S" (error "Unknown key: :docs-link.  Available keys: (:name :..."))
  internal-macroexpand-for-load((defalias 'construct-tf-module #'(lambda (input0) "Construct `TF-MODULE' using MODULE." (-let* (((&terraform-ls:Module :name :docs-link :version :source-type :dependent-modules) input0)) (make-tf-module :name name :doc-link docs-link :version version :source-type source-type :dependent-modules dependent-modules)))) t)
  load-with-code-conversion("/home/padraic/.config/emacs/.local/straight/build-..." "/home/padraic/.config/emacs/.local/straight/build-..." t t)
  require(lsp-terraform nil t)
  (if (featurep package) nil (require package nil t))
  (closure (tramp-ssh-controlmaster-options tramp-use-ssh-controlmaster-options view-inhibit-help-message lsp-help-mode-abbrev-table lsp-help-mode-syntax-table eldoc-documentation-default cl-struct-lsp--log-entry-tags cl-struct-lsp-session-tags cl-struct-lsp--workspace-tags cl-struct-lsp--registered-capability-tags lsp-mode-menu cl-struct-lsp--folding-range-tags cl-struct-lsp-watch-tags cl-struct-lsp--client-tags lsp--log-lines dap-ui-menu-items dap-auto-configure-mode t) (package) (if (featurep package) nil (require package nil t)))(lsp-terraform)
  mapc((closure (tramp-ssh-controlmaster-options tramp-use-ssh-controlmaster-options view-inhibit-help-message lsp-help-mode-abbrev-table lsp-help-mode-syntax-table eldoc-documentation-default cl-struct-lsp--log-entry-tags cl-struct-lsp-session-tags cl-struct-lsp--workspace-tags cl-struct-lsp--registered-capability-tags lsp-mode-menu cl-struct-lsp--folding-range-tags cl-struct-lsp-watch-tags cl-struct-lsp--client-tags lsp--log-lines dap-ui-menu-items dap-auto-configure-mode t) (package) (if (featurep package) nil (require package nil t))) (ccls lsp-actionscript lsp-ada lsp-angular lsp-ansible lsp-autotools lsp-awk lsp-asm lsp-astro lsp-bash lsp-beancount lsp-bufls lsp-clangd lsp-clojure lsp-cmake lsp-credo lsp-crystal lsp-csharp lsp-css lsp-cypher lsp-d lsp-dart lsp-dhall lsp-docker lsp-dockerfile lsp-elm lsp-elixir lsp-emmet lsp-erlang lsp-eslint lsp-fortran lsp-fsharp lsp-gdscript lsp-go lsp-golangci-lint lsp-gleam lsp-glsl lsp-graphql lsp-hack lsp-grammarly lsp-groovy lsp-haskell lsp-haxe lsp-idris lsp-java lsp-javascript lsp-json lsp-kotlin lsp-latex lsp-ltex ...))
  seq-do((closure (tramp-ssh-controlmaster-options tramp-use-ssh-controlmaster-options view-inhibit-help-message lsp-help-mode-abbrev-table lsp-help-mode-syntax-table eldoc-documentation-default cl-struct-lsp--log-entry-tags cl-struct-lsp-session-tags cl-struct-lsp--workspace-tags cl-struct-lsp--registered-capability-tags lsp-mode-menu cl-struct-lsp--folding-range-tags cl-struct-lsp-watch-tags cl-struct-lsp--client-tags lsp--log-lines dap-ui-menu-items dap-auto-configure-mode t) (package) (if (featurep package) nil (require package nil t))) (ccls lsp-actionscript lsp-ada lsp-angular lsp-ansible lsp-autotools lsp-awk lsp-asm lsp-astro lsp-bash lsp-beancount lsp-bufls lsp-clangd lsp-clojure lsp-cmake lsp-credo lsp-crystal lsp-csharp lsp-css lsp-cypher lsp-d lsp-dart lsp-dhall lsp-docker lsp-dockerfile lsp-elm lsp-elixir lsp-emmet lsp-erlang lsp-eslint lsp-fortran lsp-fsharp lsp-gdscript lsp-go lsp-golangci-lint lsp-gleam lsp-glsl lsp-graphql lsp-hack lsp-grammarly lsp-groovy lsp-haskell lsp-haxe lsp-idris lsp-java lsp-javascript lsp-json lsp-kotlin lsp-latex lsp-ltex ...))
  (progn (seq-do #'(lambda (package) (if (featurep package) nil (require package nil t))) lsp-client-packages) (setq lsp--client-packages-required t))
  (if (and lsp-auto-configure (not lsp--client-packages-required)) (progn (seq-do #'(lambda (package) (if (featurep package) nil (require package nil t))) lsp-client-packages) (setq lsp--client-packages-required t)))
  lsp--require-packages()
  lsp()
  (progn (remove-hook 'window-configuration-change-hook #'lsp--init-if-visible t) (lsp) t)
  (if (lsp--buffer-visible-p) (progn (remove-hook 'window-configuration-change-hook #'lsp--init-if-visible t) (lsp) t))
  lsp--init-if-visible()
  (if (lsp--init-if-visible) nil (add-hook 'window-configuration-change-hook #'lsp--init-if-visible nil t))
  (save-current-buffer (set-buffer buffer) (if (lsp--init-if-visible) nil (add-hook 'window-configuration-change-hook #'lsp--init-if-visible nil t)))
  (progn (save-current-buffer (set-buffer buffer) (if (lsp--init-if-visible) nil (add-hook 'window-configuration-change-hook #'lsp--init-if-visible nil t))))
  (if (buffer-live-p buffer) (progn (save-current-buffer (set-buffer buffer) (if (lsp--init-if-visible) nil (add-hook 'window-configuration-change-hook #'lsp--init-if-visible nil t)))))
  (closure ((buffer . #<buffer main.go>)) nil (if (buffer-live-p buffer) (progn (save-current-buffer (set-buffer buffer) (if (lsp--init-if-visible) nil (add-hook 'window-configuration-change-hook #'lsp--init-if-visible nil t))))))()
  apply((closure ((buffer . #<buffer main.go>)) nil (if (buffer-live-p buffer) (progn (save-current-buffer (set-buffer buffer) (if (lsp--init-if-visible) nil (add-hook 'window-configuration-change-hook #'lsp--init-if-visible nil t)))))) nil)
  timer-event-handler([t 0 0 0 nil (closure ((buffer . #<buffer main.go>)) nil (if (buffer-live-p buffer) (progn (save-current-buffer (set-buffer buffer) (if (lsp--init-if-visible) nil (add-hook ... ... nil t)))))) nil idle 0 nil])
```

### Steps to reproduce

Executing any language using lsp-mode fails

### System Information

https://pastebin.com/JRTTQx5D

**Comment by hlissner** [comment]
Likely emacs-lsp/lsp-mode#3577. Check the thread for proposed workarounds.

**Comment by freizl** [comment]
Im getting this error due to this changes

```
(void-variable lsp-client-packages)
```

**Comment by hlissner** [comment]
My bad. Fixed in d4357c1
